lib/LaTeXML/Engine/Base_ParameterTypes.pool.ltxml

```diff
@@ -148,11 +148,26 @@ DefParameterType('XUntil', sub {
         push(@tokens, $token); } }
     Tokens(@tokens); });
 
-# This reads a braced tokens list, expanding as it goes,
-# but expanding \the-like commands only once.
+# Simulate reading a plain argument, and then fully expanding it.
+# Similar to when \csname is used
 DefParameterType('Expanded', sub {
     my ($gullet) = @_;
-    $gullet->readBalanced(1, 0, 1); },
+    my $token = $gullet->readToken;
+    if (!$token->equals(T_BEGIN)) {    # if non-brace, wrap the token in {}, to isolate it
+      $gullet->unread($token, T_END); }
```

Comment:
Scary coding pattern. unread `T_END` then invoke readBalanced with `$require_open` set to "false".

Not to say it is wrong. Just wondering if it is scary enough to get its own named subroutine `readExpanded` in Gullet. Which can then take the `$expanded` arg as an input arg.

I think this PR uses the technique 4 times, so maybe there's some value in naming it...

Reply:
I was initially inclined to think that it *should* be (slightly) scary, rather than hidden away, since it's doing something rather peculiar. The use case is you're reading a "plain" argument, like `#1`, which would also accept a single token, and then asking it to be expanded. What if it's a macro expecting arguments?
Alternatively, it may want a helper function if there's a sensible (but likely convoluted) error recovery; At this point I don't know what that would be. (but see below)